tests/convert/test_interval.py

```diff
@@ -0,0 +1,325 @@
+        assert actual == expected
+
+    def test_to_hours_zero(self):
+
+        interval = Interval('test', 'PT0H', 'PT1H')
+        actual = interval.to_hours()
+
+        assert actual == (0, 1)
+
+    def test_to_hours_month(self):
+
+        interval = Interval('test', 'P2M', 'P3M')
+        actual = interval.to_hours()
+
+        assert actual == (1416, 2160)
+
+    def test_str(self):
+        interval = Interval('test', 'P2M', 'P3M')
+        actual = str(interval)
+        assert actual == "Interval 'test' starts at hour 1416 and ends at hour 2160"
+
+    def test_repr(self):
+        interval = Interval('test', 'P2M', 'P3M', 2011)
+        actual = repr(interval)
+        assert actual == "Interval('test', 'P2M', 'P3M', base_year=2011)"
+
+
+class TestTimeSeries:
+
+    def test_load_time_series(self, months):
+
+        data = [{'name': '1_0', 'value': 1},
+                {'name': '1_1', 'value': 1},
+                {'name': '1_2', 'value': 1},
+                {'name': '1_3', 'value': 1},
+                {'name': '1_4', 'value': 1},
+                {'name': '1_5', 'value': 1},
+                {'name': '1_6', 'value': 1},
+                {'name': '1_7', 'value': 1},
+                {'name': '1_8', 'value': 1},
+                {'name': '1_9', 'value': 1},
+                {'name': '1_10', 'value': 1},
+                {'name': '1_11', 'value': 1}]
+
+        register = TimeIntervalRegister(2010)
+        register.add_interval_set(months, 'months')
+
+        timeseries = TimeSeries(data, register)
+        actual = timeseries.names
+        expected = ['1_0', '1_1', '1_2', '1_3', '1_4', '1_5',
+                    '1_6', '1_7', '1_8', '1_9', '1_10', '1_11']
+        assert actual == expected
+
+        actual = timeseries.values
+        expected = [1] * 12
+        assert actual == expected
+
+        timeseries._convert_to_hourly_buckets()
+        actual = timeseries._hourly_values
+
+        month_days = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31]
+        month_hours = [x for x in map(lambda x: x*24, month_days)]
+        expected_results = [x for x in map(lambda x: 1/x, month_hours)]
+
+        start = 0
+        for hours, expected in zip(month_hours, expected_results):
+            expected_array = np.zeros(hours, dtype=np.float)
+            expected_array.fill(expected)
+            assert_equal(actual[start:start + hours], expected_array)
+            start += hours
+
+
+class TestTimeSeriesConversion:
+
+    def test_convert_from_month_to_seasons(self,
+                                           months,
+                                           seasons,
+                                           monthly_data):
+
+        data = monthly_data
+
+        register = TimeIntervalRegister()
+        register.add_interval_set(months, 'months')
+        register.add_interval_set(seasons, 'seasons')
+
+        timeseries = TimeSeries(data, register)
+        actual = timeseries.convert('seasons')
```

Comment:
How about switching this API around like:
```python
ts = Timeseries(data)
register.convert(ts, 'hourly_day', 'one_day') # where the params are: data, from_set_name, to_set_name
```

So a register would know how to do conversions, and a TimeSeries would just hold a set of parsed TimeIntervals.